lib/Sabberworm/CSS/Value/Value.php

```diff
@@ -68,7 +68,15 @@ public static function parseIdentifierOrFunction(ParserState $oParserState, $bIg
 		if ($oParserState->comes('(')) {
 			$oParserState->consume('(');
 			$aArguments = Value::parseValue($oParserState, array('=', ' ', ','));
-			$sResult = new CSSFunction($sResult, $aArguments, ',', $oParserState->currentLine());
+
+			if ($sResult == 'url') {
+				$sResult = new URL($aArguments, $oParserState->currentLine());
+			} else if ($sResult == 'calc') {
+				$sResult = new CalcFunction($sResult, $aArguments, ',', $oParserState->currentLine());
+			} else {
+				$sResult = new CSSFunction($sResult, $aArguments, ',', $oParserState->currentLine());
+			}
+
```

Comment:
Can we unify this with the logic from `parsePrimitiveValue` (i.e. drop the duplicate code and `URL::parse` and `CalcFunction::parse`)?